tsl/src/continuous_aggs/refresh.c

```diff
@@ -532,50 +543,44 @@ continuous_agg_refresh_internal(const ContinuousAgg *cagg,
 }
 
 /*
- * Refresh all continuous aggregates on a hypertable.
+ * Refresh a continuous aggregate on a chunk.
  *
  * The refreshing happens in a single transaction. For this to work correctly,
  * there must be no new invalidations written in the refreshed region during
- * the refresh. Therefore, the caller is responsible for proper locking to
- * ensure there are no invalidations (INSERTs, DELETEs, etc.). For instance,
- * exclusively locking the hypertable or the individual chunks covered by the
- * region would work.
+ * the refresh. Therefore, it locks exclusively the chunk to
+ * ensure there are no invalidations (INSERTs, DELETEs, etc.).
  */
-void
-continuous_agg_refresh_all(const Hypertable *const ht, const int64 start, const int64 end,
-						   const int32 chunk_id)
+Datum
+continuous_agg_refresh_chunk(PG_FUNCTION_ARGS)
 {
+	Oid cagg_relid = PG_ARGISNULL(0) ? InvalidOid : PG_GETARG_OID(0);
+	Oid chunk_relid = PG_ARGISNULL(1) ? InvalidOid : PG_GETARG_OID(1);
+	ContinuousAgg *cagg = get_cagg_by_relid(cagg_relid);
+	Chunk *chunk = ts_chunk_get_by_relid(chunk_relid, true);
```

Comment:
I think we want to validate somehow that this chunk belongs to the hypertable that the cagg is created on and raise a useful error if not. I don't see how that happens right now.

Reply:
+1